autoload/lsp/utils.vim

```diff
@@ -71,10 +71,16 @@ function! s:encode_uri(path, start_pos_encode, default_prefix) abort
     return l:prefix . l:result
 endfunction
 
+let s:path_to_uri_cache = {}
 if has('win32') || has('win64')
     function! lsp#utils#path_to_uri(path) abort
+        if has_key(s:path_to_uri_cache, a:path)
```

Comment:
have you tried using `get(s:, 'path_to_uri_cache', v:null)` instead. this could help further by reducing the call to hash function twice. same for other places.